git-publish

```diff
@@ -449,14 +449,25 @@ def menu_select(menu):
             continue
         return a
 
+def edit_content(content, suffix):
+    fd, tmpfile = tempfile.mkstemp(text=True, suffix=suffix)
+    fd_opened = os.fdopen(fd, 'wb')
+    fd_opened.write(content)
+    fd_opened.close()
+    edit(tmpfile)
+    fd_opened = open(tmpfile, "rb")
+    new_content = fd_opened.read()
+    fd_opened.close()
+    if tmpfile:
+        os.unlink(tmpfile)
```

Comment:
Please use a `try: ... finally: ...` block to catch all exceptions and unlink the file before re-raising the exception. This way the temporary file will be deleted in most cases where the program is interrupted or crashes.